### DISCUSSION TITLE
Alarm 6

### DISCUSSION THREAD

**Discussion by dennishatton** [grblHAL/core#122] [Q&A] [answered]
I've just installed grblHAL on SKR v1.4 with DRV8825 drivers.
I was previously using Smoothieware without any problem.
Now when I try homing I get Z axis to home but get an Alarm 6 on X axis
"Reset was issued during a homing cycle"
I have closed loop limit switches.
I put jumpers to ground for the control switches.

**Comment by terjeio** [comment, marked as answer]
> I have closed loop limit switches.
I put jumpers to ground for the control switches.

Is the grblHAL pin mappings the same as Smoothieware uses? If not have you rewired to match grblHAL?

The `$LIM` and `$LEV` can be used to output the current limit pins state and latest control and limit signals state. What is the output from these after the alarm is triggered?

**Reply by dennishatton** [reply]
If I disable hard limits It homes ok.
With limits enabled I am now getting ALARM 8
I set pull off to 5 
I was able to use 1 before
The ports are the same x 1.29, y 1.28, z 1.27

$LIM
[MSG:'$H'|'$X' to unlock]
[LIMITS:,,,]
ok
$G
[GC:G0 G54 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F0 S0.]
ok
$LEV
[LASTEVENTS:,Y,,,]
ok
$G
[GC:G0 G54 G17 G21 G90 G94 G49 G98 G50 M5 M9 T0 F0 S0.]
ok

**Reply by terjeio** [reply]
> If I disable hard limits It homes ok.

That is it - due to the LPC176x 3D printer boards using port 1, which is not interrupt capable, for limit inputs I had to use continuous polling to implement hard limits. And polling is not disabled during homing as it should... Try with this driver.c:

[driver.zip](https://github.com/grblHAL/core/files/8166943/driver.zip)

**Reply by dennishatton** [reply]
That seem to fix it. Thank you.
